### ISSUE TITLE
Stutter when scrolling

### ISSUE THREAD

**Issue by akeva001** [Shopify/flash-list#802]
When I scroll through the items in the FlashList, I notice slight stutters. Any help is appreciated. I am rendering posts which contain an image along with a header and footer for each <Post/>

Platform:
- iOS

**Comment by andrespch** [comment]
Hey @akeva001, can you please provide an example project that demonstrates this issue? we would then try to look into this soon.

**Comment by arthurgeron-work** [comment]
@akeva001 Can you ensure you're using the best values for `estimatedListSize` and `estimatedItemSize`?

**Comment by efstathiosntonas** [comment]
do you by any chance use expo-image? I was using my own caching image implementation and when I gave expo-image a go then I've noticed those stutters too, mostly on Android devices.

**Comment by VictorioMolina** [comment]
[efstathiosntonas](https://github.com/efstathiosntonas) I also use expo-image. With normal images and the estimatedSizes recommended by the FlashList everything is OK, but the strange behaviour comes with expo-image's `cachePolicy`. Just change to `cachePolicy="none"` where you get the buggy behaviour.

**Comment by efstathiosntonas** [comment]
@VictorioMolina I rely on cache so setting it to none serves no purpose on my case 🤷‍♂️

**Comment by tanishqk-shortwave** [comment]
FYI: we ran into this at Shortwave. We're pretty sure it's happening because `FlashList` does not maintain scroll position when items above the render window change size. So this is what happens:

* you scroll up, render window shifts
* item above renders
* item above changes size for some reason
* your scroll offset (measured from the top), stays the same but visually, it'll look like you've been jumped up.

This is a problem for views with images since, usually, they're height 0 until loaded. This was also a problem for us since we load HTML for threads, which are also size 0 until loaded. We got around it by creating a native module "AnchoredScrollView" that will keep your scroll position anchored even if items above your render window change size. Unfortunately, that means we couldn't use FlashList (we could've forked but there were other problems we ran into). I did [publish a gist](https://gist.github.com/tanishqk-shortwave/981dc521e74552c58860efa83ecf6bd7) here. Hopefully, this fix will be adopted by FlashList as well.

**Comment by GeorgiIvanovPXL** [comment]
Yeah still appearing indeed when the cachePolicy option is other than “none” if using expo image. Pretty annoying!

**Comment by pepew-le-boss** [comment]
Still the case. I have to dump expo-image or flash-list, not ideal

**Comment by chrisbianca** [comment]
@irisjae I attempted to use your patch, and found that it did fix the stuttering performance when scrolling up. However, I found that if I scrolled very quickly to the top, then the whole view would jump when I reached the top of the view.

**Comment by irisjae** [comment]
@chrisbianca What OS did you try it on? Could you post a video? You may get better results by tuning the props in the patch.

**Comment by cf913** [comment]
i managed to solve the stuttering with expo-image by setting  `allowDownscaling` to false. It's true by default.

my project displays user avatars in a long Flashlist and I resize the images at upload time already. So there's no need to downscale thumbnail sized images, and the performance improvement is quite noticeable.

From the expo image docs about the `allowDownscaling` prop:
_Whether the image should be downscaled to match the size of the view container. Turning off this functionality could negatively impact the application's performance, particularly when working with large assets. However, it would result in smoother image resizing, and end-users would always have access to the highest possible asset quality._

**Comment by naqvitalha** [comment]
Hi! 👋

We're now focusing exclusively on **FlashList v2** and **React Native's New Architecture**. This means we're closing older issues and suggestions that aren't relevant to v2.

**If this issue/suggestion is still relevant:**
- Please verify it still occurs in FlashList v2
- Open a new issue with v2-specific details
- We'll be happy to take a look!

**Resources:**
- [Migration to v2](https://shopify.github.io/flash-list/docs/migration-guide)
- [FlashList Documentation](https://shopify.github.io/flash-list/)

Thank you for your understanding! 🙏
